### ISSUE TITLE
White screen after login

### ISSUE THREAD

**Issue by WoodrowShigeru** [IsmaelMartinez/teams-for-linux#1494]
**Describe the bug**
I am getting the email and password prompts; and after that, even the question to "stay signed in" – it doesn't matter whether I click Yes or No here.

After that, there are some loading icons but eventually, I'm stuck on the white screen afterwards. (Technically, there is a white box with rounded corners and drop-shadow).

Upon opening the debug mode and refreshing, the console shows a multitude of errors.

```
[Auth]: auth_cache initialized
[Auth]: authTelemetryProvider initialized
[Auth]: [Telemetry][f9109af9-7bca-4755-aee5-b10a77f51336][auth_web_initialization] [start] - telemetry provider created for createAuthenticationServiceInstance, source: core-auth-tasks::init with shouldCreateScenario: [true]
[Auth]: authTelemetryProvider initialized
Uncaught (in promise) Error: Discover::regionGtm::InvalidResponse - peopleProfileService is missing in discover response
    at c (precompiled-web-worker-3661e62c1ac1c411.js:1:1779620)
    at precompiled-web-worker-3661e62c1ac1c411.js:1:6016565
Uncaught (in promise) Error: Discover::regionGtm::InvalidResponse - presenceUPS is missing in discover response
    at c (precompiled-web-worker-3661e62c1ac1c411.js:1:1779620)
    at precompiled-web-worker-3661e62c1ac1c411.js:1:6016565
Uncaught (in promise) Error: Discover::regionGtm::InvalidResponse - teamsAndChannelsProvisioningService is missing in discover response
    at c (precompiled-web-worker-3661e62c1ac1c411.js:1:1779620)
    at precompiled-web-worker-3661e62c1ac1c411.js:1:6016565
Uncaught (in promise) Error: Discover::regionGtm::InvalidResponse - iC3LocationService is missing in discover response
    at c (precompiled-web-worker-3661e62c1ac1c411.js:1:1779620)
    at precompiled-web-worker-3661e62c1ac1c411.js:1:6016565
Uncaught (in promise) Error: Discover::regionGtm::InvalidResponse - appsSuggestionService is missing in discover response
    at c (precompiled-web-worker-3661e62c1ac1c411.js:1:1779620)
    at precompiled-web-worker-3661e62c1ac1c411.js:1:6016565
Uncaught (in promise) Error: Discover::regionGtm::InvalidResponse - iC3LocationService is missing in discover response
    at c (precompiled-web-worker-3661e62c1ac1c411.js:1:1779620)
    at precompiled-web-worker-3661e62c1ac1c411.js:1:6016565
Failed to override MS Method
Failed to start Activity Hub by assigning Event Handlers
Failed to set Default Title
Failed to set Machine State
Failed to load resource: the server responded with a status of 404 () manifest.json:1 
```

I get the same result when visiting https://teams.live.com/v2/ directly – which seems to be what this app is doing too. (tested in Chrome, Firefox, even Edge, deactivated all browser extensions, tried in incognito mode, turned off the VPN I am behind)

Further interesting error messages I get over in the browser:

```
Unhandled error/rejection {"isTrusted":true}
GET https://teams.live.com/v2/manifest.json 404 (Not Found)

Uncaught (in promise) {
	code: "Unknown"
	context: "DataLayerClient"
	errorCode: undefined
	isExpected: false
	message: "RequireStatusFailed"
	name: "PresenceCreateEndpointJob"
	originalErrorMessage: "RequireStatusFailed"
	originalErrorReason: "RequireStatusFailed"
	reason: undefined
	serverResponse: {
		status: 500
		url: "https://presence.teams.live.com/v1/me/endpoints/"
	statusCode: 500
	subContext: "PresenceCreateEndpointJob"
}
```

Why is the manifest not reachable? And why only for me?

Honestly, I'm not sure whether there is anything you can do if it's a server-side issue. But is it truly? Maybe you can help me debug this. Just because you have the insight about how this login flow works, I mean. The last commit to this repository having been a few hours ago makes me believe you aren't getting this. It all points to a local issue with my personal machine. Otherwise, I'd find way more search results about login issues like this.

**To Reproduce**
1. Open app.
2. Enter credentials and log in.
3. → White screen / error in console.

**Expected behavior**
The user is either logged in and lands on a dashboard or landing page – or they get an error message displayed in the frontend, not dev console.

**Desktop (please complete the following information):**
 - OS: Ubuntu 24.04.1 LTS
 - Installation package: snap
 - Version: 1.11.5 stable

**Debug**

```bash
$ teams-for-linux --webDebug --logConfig={}
No config file found, using default values
all good with appLogLevels you aren't using them
all good with onlineCheckMethod you aren't using them
Initialising logger with config: "{}"
13:02:25.486 › File logging at /home/$USER/snap/teams-for-linux/715/.config/teams-for-linux/logs/main.log with level silly
13:02:25.487 › Console logging enabled with level silly
13:02:25.487 › Logger initialised
13:02:25.487 › configPath: /home/$USER/snap/teams-for-linux/715/.config/teams-for-linux
13:02:25.487 › configFile: {}
13:02:25.489 › Running under Wayland, switching to PipeWire...
dbus-send: /snap/teams-for-linux/715/lib/x86_64-linux-gnu/libdbus-1.so.3: version `LIBDBUS_PRIVATE_1.12.20' not found (required by dbus-send)

(teams-for-linux:25469): Gtk-WARNING **: 13:02:25.538: Theme parsing error: gtk.css:1422:23: 'font-feature-settings' is not a valid property name

(teams-for-linux:25469): Gtk-WARNING **: 13:02:25.541: Theme parsing error: gtk.css:3308:25: 'font-feature-settings' is not a valid property name

(teams-for-linux:25469): Gtk-WARNING **: 13:02:25.542: Theme parsing error: gtk.css:3770:23: 'font-feature-settings' is not a valid property name
[25469:1125/130225.594846:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.Secret.Service.ReadAlias: object_path= /org/freedesktop/secrets: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.334" (uid=1231611329 pid=25469 comm="/snap/teams-for-linux/715/teams-for-linux --no-san" label="snap.teams-for-linux.teams-for-linux (enforce)") interface="org.freedesktop.Secret.Service" member="ReadAlias" error name="(unset)" requested_reply="0" destination="org.freedesktop.secrets" (uid=1231611329 pid=3120 comm="/usr/bin/gnome-keyring-daemon --foreground --compo" label="unconfined")
13:02:25.633 › processArgs: [
  '/snap/teams-for-linux/715/teams-for-linux',
  '--no-sandbox',
  '--webDebug',
  '--logConfig={}'
]
13:02:25.634 › testing RegExp processArgs false
13:02:25.634 › testing RegExp processArgs false
13:02:25.635 › testing RegExp processArgs false
13:02:25.635 › testing RegExp processArgs false
13:02:25.635 › Waiting for network...
13:02:25.635 › Testing network using net.request() for https://teams.microsoft.com
13:02:25.637 › Spellchecker is disabled!

(teams-for-linux:25469): Gtk-WARNING **: 13:02:25.659: Theme parsing error: gtk.css:1422:23: 'font-feature-settings' is not a valid property name

(teams-for-linux:25469): Gtk-WARNING **: 13:02:25.661: Theme parsing error: gtk.css:3308:25: 'font-feature-settings' is not a valid property name

(teams-for-linux:25469): Gtk-WARNING **: 13:02:25.661: Theme parsing error: gtk.css:3770:23: 'font-feature-settings' is not a valid property name
libGL error: MESA-LOADER: failed to open iris (search paths /snap/teams-for-linux/715/gnome-platform/usr/lib/x86_64-linux-gnu/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open swrast (search paths /snap/teams-for-linux/715/gnome-platform/usr/lib/x86_64-linux-gnu/dri)
libGL error: failed to load driver: swrast
[25563:1125/130225.713544:ERROR:angle_platform_impl.cc(44)] Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
ERR: Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
[25563:1125/130225.713765:ERROR:gl_display.cc(497)] EGL Driver message (Critical) eglInitialize: Could not create a backing OpenGL context.
[25563:1125/130225.713841:ERROR:gl_display.cc(767)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[25469:1125/130225.715428:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.login1.Manager.Inhibit: object_path= /org/freedesktop/login1: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.354" (uid=1231611329 pid=25469 comm="/snap/teams-for-linux/715/teams-for-linux --no-san" label="snap.teams-for-linux.teams-for-linux (enforce)") interface="org.freedesktop.login1.Manager" member="Inhibit" error name="(unset)" requested_reply="0" destination="org.freedesktop.login1" (uid=0 pid=1385 comm="/usr/lib/systemd/systemd-logind" label="unconfined")
[25563:1125/130225.717043:ERROR:angle_platform_impl.cc(44)] Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
ERR: Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
[25563:1125/130225.717212:ERROR:gl_display.cc(497)] EGL Driver message (Critical) eglInitialize: Could not create a backing OpenGL context.
[25563:1125/130225.717314:ERROR:gl_display.cc(767)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[25563:1125/130225.717384:ERROR:gl_display.cc(801)] Initialization of all EGL display types failed.
[25563:1125/130225.717457:ERROR:gl_ozone_egl.cc(26)] GLDisplayEGL::Initialize failed.
[25563:1125/130225.721877:ERROR:angle_platform_impl.cc(44)] Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
ERR: Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
[25563:1125/130225.721977:ERROR:gl_display.cc(497)] EGL Driver message (Critical) eglInitialize: Could not create a backing OpenGL context.
[25563:1125/130225.722040:ERROR:gl_display.cc(767)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[25563:1125/130225.724253:ERROR:angle_platform_impl.cc(44)] Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
ERR: Display.cpp:1083 (initialize): ANGLE Display::initialize error 12289: Could not create a backing OpenGL context.
[25563:1125/130225.724407:ERROR:gl_display.cc(497)] EGL Driver message (Critical) eglInitialize: Could not create a backing OpenGL context.
[25563:1125/130225.724446:ERROR:gl_display.cc(767)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[25563:1125/130225.724485:ERROR:gl_display.cc(801)] Initialization of all EGL display types failed.
[25563:1125/130225.724519:ERROR:gl_ozone_egl.cc(26)] GLDisplayEGL::Initialize failed.
[25563:1125/130225.726213:ERROR:viz_main_impl.cc(181)] Exiting GPU process due to errors during initialization
MESA-LOADER: failed to open iris (search paths /snap/teams-for-linux/715/gnome-platform/usr/lib/x86_64-linux-gnu/dri)
failed to load driver: iris
MESA-LOADER: failed to open kms_swrast (search paths /snap/teams-for-linux/715/gnome-platform/usr/lib/x86_64-linux-gnu/dri)
failed to load driver: kms_swrast
MESA-LOADER: failed to open swrast (search paths /snap/teams-for-linux/715/gnome-platform/usr/lib/x86_64-linux-gnu/dri)
failed to load swrast driver
13:02:25.928 › Network test successful with method https
13:02:26.305 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[25469:1125/130226.308005:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not part of the idle inhibition specification: https://specifications.freedesktop.org/idle-inhibit-spec/latest/
13:02:26.661 › did-frame-finish-load {} true
13:02:26.661 › did-finish-load
13:02:28.503 › Badge count set to '0'
13:02:28.748 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
13:02:28.920 › did-frame-finish-load {} true
13:02:28.920 › did-finish-load
13:02:29.763 › Badge count set to '0'
13:02:29.896 › did-frame-finish-load {} false
13:02:38.788 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
```

I'm noticing this `org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient;` which seems odd. I had run `echo 'kernel.apparmor_restrict_unprivileged_userns = 0' | sudo tee /etc/sysctl.d/20-apparmor-donotrestrict.conf` a few months ago in order to install Docker, which had deactivated the armor feature altogether, to my understanding.

**Additional information**
According to [this guide](https://idroot.us/install-microsoft-teams-ubuntu-24-04/), I could supposedly install a *.DEB file instead from [the official download page](https://www.microsoft.com/en-us/microsoft-teams/download-app). But I see no *.DEB link there – only a link that redirects me to https://teams.live.com/v2/

**Comment by IsmaelMartinez** [comment]
Hello @WoodrowShigeru,

There seem to be a few different problems you are reporting in here. I will try to answer them:

> I get the same result when visiting https://teams.live.com/v2/ directly – which seems to be what this app is doing too. (tested in Chrome, Firefox, even Edge, deactivated all browser extensions, tried in incognito mode, turned off the VPN I am behind)

Unfortunately if teams is not working on the browser, it will not be working in the app (as the app wraps around the browser).

Some colleagues today had problems with Microsoft, so hopefully that part is sorted. Otherwise try restarting your machine. 

> I'm noticing this org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; which seems odd. I had run echo 'kernel.apparmor_restrict_unprivileged_userns = 0' | sudo tee /etc/sysctl.d/20-apparmor-donotrestrict.conf a few months ago in order to install Docker, which had deactivated the armor feature altogether, to my understanding.

This is a change Ubuntu introduced to improve security. The best option we have found so far is to setup an apparmour profile. More details in https://github.com/IsmaelMartinez/teams-for-linux/issues/1426#issuecomment-2453340003 

> Additional information
According to [this guide](https://idroot.us/install-microsoft-teams-ubuntu-24-04/), I could supposedly install a *.DEB file instead from [the official download page](https://www.microsoft.com/en-us/microsoft-teams/download-app). But I see no *.DEB link there – only a link that redirects me to https://teams.live.com/v2/

The README has info on how to install the deb package.
https://github.com/IsmaelMartinez/teams-for-linux?tab=readme-ov-file#debianubuntu-and-other-derivatives . You can also just get the files from the releases section. 

Hope helps and I do hope the "white screen" issue is sorted. I don't think is the same as this https://github.com/IsmaelMartinez/teams-for-linux/blob/main/KNOWN_ISSUES.md#blank-page (that basically was related to cache), but putting the link just in case.

**Comment by WoodrowShigeru** [comment]
Thanks for the super fast reply.

**Restart:** Unfortunately, restarting the machine doesn't work. I've had this problem … I've been trying to install Teams since last Thursday. For a couple of days, in other words. I just gave it a try, anyways – but no diff.

Are your colleagues / you really able to load https://teams.live.com/v2/manifest.json without a 404 response?

**AppArmor:** I just tried the apparmor fix also +restart. Unfortunately, that didn't change anything either. Strangely enough, even that error message is maintained … Maybe I created the file at the wrong place.

**Cache:** With my snap installation, I see this structure:

```
~/snap/teams-for-linux/current/.config/teams-for-linux/Partitions/teams-4-linux/
├── Cache
├── Code Cache
…
```

But no "Application Cache". (Probably just outdated notes; not the cause of this issue). Deleting those doesn't help either.

**DEB:**

This thread [Where have the Linux DEB packages for Teams gone?](https://learn.microsoft.com/en-us/answers/questions/440594/where-have-the-linux-deb-packages-for-teams-gone) has a helpful comment posting [an archived directory with old versions](https://web.archive.org/web/20221203104824/https://packages.microsoft.com/repos/ms-teams/pool/main/t/teams/) from which I could install an older version of the old client.

```bash
# install 
$ sudo dpkg -i teams_1.4.00.7556_amd64.deb 

# fix dependency issues.
$ sudo apt-get install --fix-broken

# try again.
$ sudo dpkg -i teams_1.4.00.7556_amd64.deb 
```

And that finally worked. _(Aside from the next roadblock "Someone has already set up Teams for your organization" which I have to figure out with my workplace admins)_

Thanks for helping me out, and helping me get this figured out.

**Comment by IsmaelMartinez** [comment]
fyi, that package is the non supported version by Microsoft that they stop supporting a few years ago. Use it at your own risk

**Comment by WoodrowShigeru** [comment]
I know but the PWA solution doesn't work at all for me. With this, at least, I get past the login screen.

Will have to wait for the first call to know for sure whether this works all the way, though.

**Comment by IsmaelMartinez** [comment]
I was meaning, you should be able to use our app instead of an app that dropper support more than 2 years ago.

**Comment by WoodrowShigeru** [comment]
"should be" … If the old client works better than the new one (which it did for some users), then I'll likely not use the version that leaves me stuck with a whitescreen.

But well, neither works, as it turns out. This older client doesn't even have a settings sidebar, or any menu whatsoever; and it won't let me join any meeting (no button) or open the app when I click on a link (e.g. from an email).

Strangely enough, this link format lets me join a meeting in the browser, and set up the camera (probably because I'm signing in as a guest): (I've altered the IDs because security)

https://teams.microsoft.com/l/meetup-join/19%3ameeting_AbcAbcAbcAbcAbcAbc00Y2JmLTk2YWUtOTZjY2I0ZjIyM2Ex%40thread.v2/0?context=%7b%22Tid%22%3a%2211aa30a8-ad2e-4fed-8181-32ab739426bd%22%2c%22Oid%22%3a%224d135f2e-3bcf-43b2-b7c1-fa1234533de5%22%7d

… whereas this doesn't:
https://teams.live.com/meet/12345957558939?p=AbcabcZ2eV2teFE24Q

It doesn't matter at this point – we've decided that I abandon trying to insteall Teams on linux and join that one meeting that I need it for another way.
